components/Video/Video.js

```diff
@@ -57,23 +56,34 @@ class Video extends PureComponent {
     },
   };
 
+  constructor(props) {
+    super(props);
+
+    const { source, playerParams } = props;
+    this.sourceReader = new VideoSourceReader(source.uri, playerParams);
+  }
+
   render() {
     const {
       width,
       height,
-      source,
       style,
-      playerParams,
       poster,
     } = this.props;
 
+    // https://github.com/vimeo/player.js/issues/514
+    // Vimeo player crashes the app, if played in full screen portrait mode and if user
+    // tries to navigate back using Android hardware back button. Disableing full screen
+    // option for Vimeo until their player is fixed.
+    const isYoutubeVideo = this.sourceReader.isYouTube;
```

Comment:
Why not add `|| Platform.OS === 'ios'`?

You said it works on iOS without issue, it would be good to then allow iOS devices to display it fullscreen.

If this requires the URL parameter in `VideoSourceReader.js` (`playsinline=0`), then you can add a check for device OS there and choose whether to add the parameter or not.

Reply:
iOS plays videos in full-screen by default, on video start, it goes full-screen automatically.

Reply:
Perfect, approving.

Reply:
Should've given more details in the PR description again, I guess.

iOS plays both Vimeo and Youtube videos perfectly. It always did.
It's because they're playing videos in a native player.
On iOS, when the user starts the video, the player goes full-screen and plays. Player controls are native.
So, almost none of these params are "needed" by iOS.

On Android devices, videos are played in Vimeo/Youtube players, inside WebView.
Player controls are theirs, all user inputs (including pressing player controls) are handled by the video player itself.
That's why we are disabling Vimeo Android full-screen - Vimeo player has a bug.

Passed `playsinline=0` was added only because we were trying to enable Vimeo full-screen video play in the previous Video component PR.
We haven't had the option to go full-screen before.
So, `playsinline=0`would cause the Vimeo player to go full-screen automatically when the user started playing the video.
Now, that we're "rollbacking" these changes Vimeo player should play inline again, without the option for full-screen, like it was before.
On iOS, this does nothing, it works perfectly itself.